### PR TITLE
Add jobs for testing TPMv2

### PR DESCRIPTION
**Pull request by nmagnezi**
Co-authored-by: Osher De Paz odepaz@redhat.com

### CODE REVIEW

File: ci-operator/config/openshift/assisted-test-infra/openshift-assisted-test-infra-master.yaml [hunk 1]
```diff
@@ -133,7 +153,7 @@ tests:
       ASSISTED_CONFIG: |
         TEST_FUNC=test_olm_operator
       TEST_SUITE: full
-    workflow: baremetalds-assisted
+    workflow: baremetalds-assisted-install
```
Comment (osherdp) [inline ci-operator/config/openshift/assisted-test-infra/openshift-assisted-test-infra-master.yaml:156]: you can remove all those changes, no need for you to see failures related to them

File: ci-operator/jobs/openshift/assisted-service/openshift-assisted-service-master-presubmits.yaml [hunk 2]
```diff
@@ -1584,6 +1584,81 @@ presubmits:
         secret:
           secretName: result-aggregator
     trigger: (?m)^/test( | .* )e2e-metal-assisted-single-node,?($|\s.*)
+  - agent: kubernetes
+    always_run: true
```
Comment (osherdp) [inline ci-operator/jobs/openshift/assisted-service/openshift-assisted-service-master-presubmits.yaml:1588]:
```suggestion
    always_run: false
```

File: ci-operator/jobs/openshift/assisted-test-infra/openshift-assisted-test-infra-master-periodics.yaml [hunk 3]
```diff
@@ -1008,6 +1008,76 @@ periodics:
...
+  - base_ref: master
+    org: openshift
+    repo: assisted-test-infra
+  labels:
+    ci-operator.openshift.io/cloud: packet
+    ci-operator.openshift.io/cloud-cluster-profile: packet-assisted
+    ci.openshift.io/generator: prowgen
+    pj-rehearse.openshift.io/can-be-rehearsed: "true"
+  name: periodic-ci-openshift-assisted-test-infra-master-e2e-metal-assisted-tpmv2-periodic
+  spec:
```
Comment (osherdp) [inline ci-operator/jobs/openshift/assisted-test-infra/openshift-assisted-test-infra-master-periodics.yaml:1025]:
```suggestion
  reporter_config:
    slack:
      channel: '#single-node-ci'
      job_states_to_report:
      - success
      - failure
      - error
      report_template: '{{ if eq .Status.State "success" }} :green_jenkins_circle:
        {{ else }} :red_jenkins_circle: {{ end }} Job *{{.Spec.Job}}* ended with *{{.Status.State}}*.
        <{{.Status.URL}}|View logs>'
  spec:
```

File: ci-operator/jobs/openshift/assisted-test-infra/openshift-assisted-test-infra-master-presubmits.yaml [hunk 4]
```diff
@@ -1028,6 +1028,79 @@ presubmits:
         secret:
           secretName: result-aggregator
     trigger: (?m)^/test( | .* )e2e-metal-assisted-single-node,?($|\s.*)
+  - agent: kubernetes
+    always_run: true
```
Comment (osherdp) [inline ci-operator/jobs/openshift/assisted-test-infra/openshift-assisted-test-infra-master-presubmits.yaml:1032]:
```suggestion
    always_run: false
```

**Review by osherdp — CHANGES_REQUESTED** [review]

File: ci-operator/config/openshift/assisted-test-infra/openshift-assisted-test-infra-master.yaml (hunk 1 above)
Comment (nmagnezi) [reply ci-operator/config/openshift/assisted-test-infra/openshift-assisted-test-infra-master.yaml]: done

File: ci-operator/jobs/openshift/assisted-service/openshift-assisted-service-master-presubmits.yaml (hunk 2 above)
Comment (nmagnezi) [reply ci-operator/jobs/openshift/assisted-service/openshift-assisted-service-master-presubmits.yaml]: done

File: ci-operator/jobs/openshift/assisted-test-infra/openshift-assisted-test-infra-master-periodics.yaml (hunk 3 above)
Comment (nmagnezi) [reply ci-operator/jobs/openshift/assisted-test-infra/openshift-assisted-test-infra-master-periodics.yaml:1035]: done

File: ci-operator/jobs/openshift/assisted-test-infra/openshift-assisted-test-infra-master-presubmits.yaml (hunk 4 above)
Comment (nmagnezi) [reply ci-operator/jobs/openshift/assisted-test-infra/openshift-assisted-test-infra-master-presubmits.yaml:1032]: done

File: ci-operator/jobs/openshift/assisted-test-infra/openshift-assisted-test-infra-master-periodics.yaml [hunk 5]
```diff
@@ -1008,6 +1008,86 @@ periodics:
...
+    repo: assisted-test-infra
+  labels:
+    ci-operator.openshift.io/cloud: packet
+    ci-operator.openshift.io/cloud-cluster-profile: packet-assisted
+    ci.openshift.io/generator: prowgen
+    pj-rehearse.openshift.io/can-be-rehearsed: "true"
+  name: periodic-ci-openshift-assisted-test-infra-master-e2e-metal-assisted-tpmv2-periodic
+  reporter_config:
+    slack:
+      channel: '#single-node-ci'
```
Comment (osherdp) [inline ci-operator/jobs/openshift/assisted-test-infra/openshift-assisted-test-infra-master-periodics.yaml:1027]: sorry, I made a mistake
it should be like this:
```suggestion
      channel: '#assisted-deployment-ci'
```

File: ci-operator/jobs/openshift/assisted-test-infra/openshift-assisted-test-infra-master-periodics.yaml [hunk 6]
```diff
@@ -1008,6 +1008,86 @@ periodics:
...
+      job_states_to_report:
+      - success
```
Comment (osherdp) [inline ci-operator/jobs/openshift/assisted-test-infra/openshift-assisted-test-infra-master-periodics.yaml:1029]: this another mistake I've made
```suggestion
```

**Review by osherdp — CHANGES_REQUESTED** [review]

File: ci-operator/jobs/openshift/assisted-test-infra/openshift-assisted-test-infra-master-periodics.yaml (hunk 6 above)
Comment (nmagnezi) [reply ci-operator/jobs/openshift/assisted-test-infra/openshift-assisted-test-infra-master-periodics.yaml:1029]: good catch.

File: ci-operator/jobs/openshift/assisted-test-infra/openshift-assisted-test-infra-master-periodics.yaml (hunk 5 above)
Comment (nmagnezi) [reply ci-operator/jobs/openshift/assisted-test-infra/openshift-assisted-test-infra-master-periodics.yaml]: done

File: ci-operator/jobs/openshift/assisted-test-infra/openshift-assisted-test-infra-master-periodics.yaml (hunk 6 above)
Comment (nmagnezi) [reply ci-operator/jobs/openshift/assisted-test-infra/openshift-assisted-test-infra-master-periodics.yaml]: done

**Review by osherdp — APPROVED** [review]
Looks great!

**Pull request merged**